react-components/src/components/Checkout/editableForm.js

```diff
@@ -66,23 +65,16 @@ const EditableForm = props => {
         dispatch({ type: 'endEditing' });
     }, [dispatch]);
 
-    const handleSubmitAddressForm = useCallback(
-        formValues => {
-            setShippingAddressesOnCart({ variables: { cartId: cartId, countryCode: 'US', ...formValues } })
-                .catch(error => {
-                    cartDispatch({ type: 'error', error: error.toString() });
-                })
-                .finally(() => {
-                    cartDispatch({ type: 'endLoading' });
-                });
-            cartDispatch({ type: 'beginLoading' });
-            cartDispatch({ type: 'endEditing' });
-            if (!isSignedIn) {
-                setGuestEmailOnCart({ variables: { cartId: cartId, email: formValues.email } });
-            }
-        },
-        [dispatch, setShippingAddressesOnCart]
-    );
+    const handleSubmitAddressForm = async formValues => {
+        cartDispatch({ type: 'beginLoading' });
```

Comment:
Regarding the error handling. Yes there should be proper error handling. You could do something like this:

```js
cartDispatch({ type: 'beginLoading' });
try {
    await (setShippingAddressesOnCart({ variables: { cartId: cartId, countryCode: 'US', ...formValues } }));
    await (getCartDetails({ cartDetailsQuery, dispatch: cartDispatch, cartId }));
} catch(err) {
    cartDispatch({ type: 'error', error: err.toString() });
} finally {
    cartDispatch({ type: 'endLoading' });
}
```